pubsub/subscription.go

```diff
@@ -256,6 +256,14 @@ type SubscriptionConfig struct {
 	// FAILED_PRECONDITION. If the subscription is a push subscription, pushes to
 	// the endpoint will not be made.
 	Detached bool
+
+	// TopicMessageRetentionDuration indicates the minimum duration for which a message is retained
+	// after it is published to the subscription's topic. If this field is set,
+	// messages published to the subscription's topic in the last
+	// `TopicMessageRetentionDuration` are always available to subscribers.
+	//
+	// This field is set only in responses from the server and otherwise ignored.
+	TopicMessageRetentionDuration time.Duration
```

Comment:
Noob question: how is this different from the `RetentionDuration time.Duration` field? It might be worth documenting the difference.

Reply:
This is essentially the same thing but you can have both set and the maximum value will be applied (https://cloud.google.com/pubsub/docs/replay-overview#configuring_message_retention).

I'll add this as a comment

Reply:
SGTM